File: js/top-k-frequent-elements.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function(nums, k) {
    const m = {}
    nums.forEach(x => m[x] = (m[x] || 0) + 1)
    
    const st = [];
    for (let x in m) {
        st.push({ num: x, cnt: m[x] })
    }
    
    st.sort((a, b) => b.cnt - a.cnt)
    const res = st.slice(0, k).map(x => x.num);
    return res;
};

File: java/top-k-frequent-elements.java
class Solution {
    public List<Integer> topKFrequent(int[] nums, int k) {
        HashMap<Integer, Integer> m = new HashMap();
        
        for (int x: nums) {
            m.put(x, m.getOrDefault(x, 0) + 1);
        }
        
        PriorityQueue<Integer> st = new PriorityQueue<Integer>((x, y) -> m.get(y)-m.get(x));
        
        for (int x: m.keySet()) st.add(x);
        
        List<Integer> res = new ArrayList();
        
        while (k-- > 0) res.add(st.poll());
        
        return res;
    }
}